### PR TITLE
Standardize backend requests

### PR DESCRIPTION
**Pull request by emveepee**
Move XML validation to single location for consistent error checking on backend requests.   Remove static strings from function calls for readability.

### CODE REVIEW

**Comment by emveepee** [comment]
The first commit has a lot of change because one level of if indenting has been removed in much of the code by moving xml processing to the calling function and returning and XML document rather then strings.  

@phunkyfish  is there any advantage to changing these calls to pass doc by reference?  Currently they follow this template

```
  tinyxml2::XMLDocument doc;
  if (m_request.DoMethodRequest("recording.list&filter=all", doc) == tinyxml2::XML_SUCCESS)
```

**Comment by phunkyfish** [comment]
Yes, always pass by reference where possible as it will avoid unnecessary copying.

File: src/pvrclient-nextpvr.cpp [hunk 1]
```diff
@@ -660,8 +660,9 @@ bool cPVRClientNextPVR::CanSeekStream(void)
 {
   if (IsServerStreaming())
   {
+    //  function is not called for recordings
```
Comment (phunkyfish) [inline src/pvrclient-nextpvr.cpp]: The comment conflicts with the code here. Maybe expand so it is clear.

Comment (emveepee) [reply src/pvrclient-nextpvr.cpp]: The documentation is not clear on how recordings deal with this.  Perhaps the best solution for now is to remove recording logic from that function entirely and assume it applies only to live tv to be consistent with other addons.

Comment (phunkyfish) [reply src/pvrclient-nextpvr.cpp]: Can seek stream is not for recordings at all. At least I never thought it was.

Comment (emveepee) [reply src/pvrclient-nextpvr.cpp]: Semantics I guess.  Here https://codedocs.xyz/AlwinEsch/kodi/group__cpp__kodi__addon__pvr___streams.html identifies 3 stream types.  If the function only applies to TV is probably should be under TV.

Comment (phunkyfish) [reply src/pvrclient-nextpvr.cpp]: Agreed the docs are not clear at all on that. Could you PR an update so we clean that up for matrix?

I think being specific even where it is currently and just specifying live only would be ok.

Comment (emveepee) [reply src/pvrclient-nextpvr.cpp]: NP I will have a look.

**Review by phunkyfish — APPROVED** [review]
LGTM

**Pull request merged**